packages/ui-number-input/src/NumberInput/README.md

```diff
@@ -10,137 +10,283 @@ This example handles arrow buttons, up/down arrow keys, and typing into
+    const invalidNumber = (value) => {
+      setMessages([
+        { text: `'${value}' is not a valid number.`, type: 'error' }
+      ])
+    }
+
+    const required = () => {
+      setMessages([{ text: 'This is required.', type: 'error' }])
+    }
+
+    const toggleDisabled = (event) => {
+      setDisabled((prevDisabled) => !prevDisabled)
+    }
+
+    const toggleInline = (event) => {
+      setInline((prevInline) => !prevInline)
+    }
+
+    const toggleReadOnly = (event) => {
+      setReadOnly((prevReadOnly) => !prevReadOnly)
+    }
+
+    const toggleShowArrows = (event) => {
+      setShowArrows((prevShowArrows) => !prevShowArrows)
+    }
 
-  render () {
     return (
       <FormFieldGroup description="Controlled NumberInput example">
         <Checkbox
-          checked={this.state.showArrows}
+          checked={showArrows}
           label="Show arrows"
-          onChange={this.toggleShowArrows}
+          onChange={toggleShowArrows}
         />
         <Checkbox
-          checked={this.state.disabled}
+          checked={disabled}
           label="Disabled"
-          onChange={this.toggleDisabled}
+          onChange={toggleDisabled}
         />
         <Checkbox
-          checked={this.state.readOnly}
+          checked={readOnly}
           label="Read only"
-          onChange={this.toggleReadOnly}
+          onChange={toggleReadOnly}
         />
         <Checkbox
-          checked={this.state.inline}
+          checked={inline}
           label="Inline display"
-          onChange={this.toggleInline}
+          onChange={toggleInline}
         />
         <NumberInput
-          renderLabel={`How old are you? (${this.MIN}-${this.MAX})`}
-          display={this.state.inline ? 'inline-block' : null}
-          messages={this.state.messages}
-          onBlur={this.handleBlur}
-          onChange={this.handleChange}
-          onDecrement={this.handleDecrement}
-          onIncrement={this.handleIncrement}
+          renderLabel={`How old are you? (${MIN}-${MAX})`}
+          display={inline ? 'inline-block' : null}
+          messages={messages}
+          onBlur={handleBlur}
+          onChange={handleChange}
+          onDecrement={handleDecrement}
+          onIncrement={handleIncrement}
           placeholder="Age (in years)"
-          interaction={this.state.disabled
-            ? 'disabled'
-            : this.state.readOnly ? 'readonly' : 'enabled'
+          interaction={
+            disabled ? 'disabled' : readOnly ? 'readonly' : 'enabled'
           }
           isRequired
-          showArrows={this.state.showArrows}
-          value={this.state.value}
+          showArrows={showArrows}
+          value={value}
         />
       </FormFieldGroup>
     )
   }
-}
 
-// IE11 doesn't support Number.isInteger
-function isInteger (number) {
-  return Math.floor(number) === number
-}
+  // IE11 doesn't support Number.isInteger
+  function isInteger(number) {
+    return Math.floor(number) === number
+  }
```

Comment:
Can you please remove this part? We dont care about IE11 anymore

Reply:
I removed the IE11 workaround